pkg/workload/workload.go

```diff
@@ -604,6 +604,26 @@ func QueuedWaitTime(wl *kueue.Workload, clock clock.Clock) time.Duration {
 	return clock.Since(queuedTime)
 }
 
+// PodsReadyToPreemptedTime is the amount of time it takes a workload's pods running to getting preempted.
+// This measures runtime of workloads that do not run to completion (ie are preempted).
+func PodsReadyToPreemptedTime(wl *kueue.Workload) *time.Duration {
+	var podsReady *time.Time
+	var preempted *time.Time
+	if c := apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadPodsReady); c != nil {
```

Comment:
Is this the correct status to look for? We need to know the time between when a plain pod is running to eviction. I want to make sure this status isn't specific to kueue's all-or-nothing feature.

Reply:
Ah, I see, in that case "Admitted" only tells you when Kueue admitted a workload. However, since then kube-scheduler still needs to schedule  the pods + the pods need to start running, which can take time in case of AI workloads to for example pull the image.

In that case, it seems you are looking for the time when "PodsReady=true" condition until Evicted. In that case I would suggest to call the metric "WorkloadsWithPodsReadyToEvictedTimeSeconds`, wdyt?